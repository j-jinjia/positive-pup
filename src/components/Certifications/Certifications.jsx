import "./Certifications.scss";
import internationalBadge from "../../assets/images/certification_badges/ICAN.png";
import ppgBadge from "../../assets/images/certification_badges/ppg2022.png";
import ukBehaviour from "../../assets/images/certification_badges/behave.png";
import victoriaStillwell from "../../assets/images/certification_badges/VSWA.png";
import petAdvocacy from "../../assets/images/certification_badges/PIAI.png";

const Certifications = ({ showText, isSmaller }) => {
  let certificationsBadgeClassName = "certifications__badge";
  if (isSmaller)
    certificationsBadgeClassName += " certifications__badge--small";

  return (
    <div className="certifications">
      {showText && (
        <>
          <h2 className="certifications__header">
            You’re in professional hands!
          </h2>
          <p className="certifications__description">
            I’m a well-established, certified trainer in Dog behaviour &#38;
            training, with over{" "}
            <span className="certifications__description--highlighted-text">
              4.5 years of experience
            </span>
            . To find out more information about any of my certifications,
            please click on one of the logos.
          </p>
        </>
      )}
      <div className="certifications__badge-container">
        <a
          href="https://companionanimal.network/"
          rel="noreferrer"
          target="_blank"
        >
          <img
            src={internationalBadge}
            className={certificationsBadgeClassName}
            alt="international companion animal network certification badge"
          />
        </a>

        <a
          href="https://www.petadvocacy.info/"
          rel="noreferrer"
          target="_blank"
        >
          <img
            src={petAdvocacy}
            className={certificationsBadgeClassName}
            alt="Pet Industry Advocacy International badge"
          />{" "}
        </a>
        <a
          href="https://www.vsdogtrainingacademy.com/"
          rel="noreferrer"
          target="_blank"
        >
          <img
            src={victoriaStillwell}
            className={certificationsBadgeClassName}
            alt="Victoria Stillwell certification badge"
          />
        </a>

        <a href="https://ukdogcharter.org/" rel="noreferrer" target="_blank">
          <img
            src={ukBehaviour}
            className={certificationsBadgeClassName}
            alt="UK dog behaviour and training charter certification badge"
          />
        </a>
        <a
          href="https://www.petprofessionalguild.com/"
          rel="noreferrer"
          target="_blank"
        >
          <img
            src={ppgBadge}
            className={certificationsBadgeClassName}
            alt="the pet professional guild certification badge"
          />
        </a>
      </div>
    </div>
  );
};

export default Certifications;
