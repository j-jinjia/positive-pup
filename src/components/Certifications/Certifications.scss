@use "../../styles/variables" as *;

.certifications {
  display: flex;
  align-items: center;
  flex-direction: column;
  padding: 2rem;

  &__header {
    font-size: $font-size-h2;
    margin-bottom: 2rem;
    line-height: 1.1;
  }

  &__description {
    margin-bottom: 2rem;
    line-height: 2;
    font-size: 1.25rem;
    color: $colour-tertiary;

    &--highlighted-text {
      color: $colour-primary;
      font-weight: 700;
      text-decoration: underline;
      text-underline-offset: 0.2rem;
      font-size: 1.25rem;
    }
  }

  &__badge-container {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    flex-wrap: wrap;
    max-height: 578px;
    .certifications__badge {
      margin: 0px 10px;
      width: 60px;
    }
  }
}

@media screen and (min-width: 460px) {
  .certifications__badge-container {
    justify-content: space-around;

    &__img {
      margin: 0px 19px;
      width: auto;
    }
  }
}

@media screen and (min-width: $tablet-breakpoint) {
  .certifications {
    text-align: center;
  }
  .certifications__badge-container {
    .certifications__badge {
      margin: 0px 19px;
      width: 5.7rem;
    }
  }
}

@media screen and (min-width: $desktop-breakpoint) {
  .certifications {
    max-width: 39.75rem;
  }
  .certifications__badge-container {
    width: 39.75rem;
    .certifications__badge {
      width: 100px;
    }
  }
}
