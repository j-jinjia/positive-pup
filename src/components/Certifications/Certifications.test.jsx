import { customRender } from "../../utils/testUtils";
import Certifications from "./Certifications";

it("should render the home page", () => {
  const { container } = customRender(<Certifications />);

  expect(container).toMatchSnapshot();
});
