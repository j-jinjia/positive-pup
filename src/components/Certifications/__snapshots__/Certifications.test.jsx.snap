// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the home page 1`] = `
<div>
  <div
    class="certifications"
  >
    <div
      class="certifications__badge-container"
    >
      <a
        href="https://www.vsdogtrainingacademy.com/"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Victoria Stillwell certification badge"
          class="certifications__badge"
          src="victoria_stillwell.png"
        />
      </a>
      <a
        href="https://companionanimal.network/"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="international companion animal network certification badge"
          class="certifications__badge"
          src="international_companion.png"
        />
      </a>
      <a
        href="https://ukdogcharter.org/"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="UK dog behaviour and training charter certification badge"
          class="certifications__badge"
          src="UK_dog_behaviour.png"
        />
      </a>
      <a
        href="https://www.petprofessionalguild.com/"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="the pet professional guild certification badge"
          class="certifications__badge"
          src="PPG_Member_Badge.png"
        />
      </a>
    </div>
  </div>
</div>
`;
